### PR TITLE
FT: Add recent metrics route

### PR DESCRIPTION
**Pull request by bennettbuchanan**
Address #52.

### CODE REVIEW

File: src/handlers/metricsHandlers.js [hunk 1]
```diff
@@ -20,6 +29,16 @@ export class BucketsHandler {
...
         return Buckets.getTypesMetrics(utapiRequest, cb);
     }
+
+    /**
+    * List metrics for the most recent 15 minutes of a given list of buckets
+    * @param {UtapiRequest} utapiRequest - UtapiRequest instance
+    * @param {callback} cb - callback
+    * @return {undefined}
+    */
+    static listRecentMetrics(utapiRequest, cb) {
```
Comment (rahulreddy) [inline src/handlers/metricsHandlers.js]: These handlers duplicate a lot of code. Can you see if you can use one class with a static method and parameterize the metric level (service/accounts/buckets/users)

**Comment by bennettbuchanan** [comment]
@rahulreddy I am wondering now if it would be helpful to instead list metrics beginning from the previous fifteen minute timestamp (i.e., if it's 6:35 when the request is made, then list metrics starting from 6:15). This would avoid a case where the listing of recent metrics would _seemingly_ not show the most recent metrics. For example, a user puts an object at 6:29, and lists recent metrics at 6:30.

**Comment by bennettbuchanan** [comment]
@ironman-machine

**Comment by bennettbuchanan** [comment]
@ironman-machine

File: src/validators/listMetrics.js [hunk 2]
```diff
@@ -4,11 +4,13 @@ import Validator from './Validator';
...
  * @param {object} dict - Input fields for route
+ * @param {boolean} [isRecentListing] - `true` if the route is for listing
+ * recent metrics, otherwise `undefined`
  * @return {Validator} Return the created validator
  */
-function getValidator(metricType, dict) {
-    const obj = {};
-    obj.timeRange = true;
+function getValidator(metricType, dict, isRecentListing) {
+    // Do not validate time ranges for recent listings because Utapi creates it
```
Comment (electrachong) [inline src/validators/listMetrics.js]: Just checking the comment was finished? Utapi creates the time ranges you mean?

File: src/validators/listMetrics.js [hunk 3]
```diff
@@ -4,11 +4,13 @@ import Validator from './Validator';
  * Get the validator for a particular metric type
  * @param {string} metricType - metric type (e.g., 'buckets', 'accounts')
  * @param {object} dict - Input fields for route
+ * @param {boolean} [isRecentListing] - `true` if the route is for listing
+ * recent metrics, otherwise `undefined`
```
Comment (electrachong) [inline src/validators/listMetrics.js]: By recent listings, do you want to specify within the most recent 15 minutes?

File: src/handlers/metricsHandlers.js [hunk 4]
```diff
@@ -67,6 +98,17 @@ export class ServiceHandler {
         const Accounts = new ListMetrics('service', service);
         return Accounts.getTypesMetrics(utapiRequest, cb);
     }
+
+    /**
+    * List metrics for the most recent 15 minutes of a given list of accounts
```
Comment (electrachong) [inline src/handlers/metricsHandlers.js]: given service?

**Comment by bennettbuchanan** [comment]
@ironman-machine
SCALITY_INTEGRATION_BRANCH=ft/S3C-78-add-recent-metrics-route-tests
try

**Comment by bennettbuchanan** [comment]
I've increased the range to start from the second most recent normalized timestamp (i.e., the definition of "recent" is now a time range up to 29:59:999).

**Comment by bennettbuchanan** [comment]
@ironman-machine try

**Comment by bennettbuchanan** [comment]
e2e passed

**Pull request merged**